thriftpy2/http.py

```diff
@@ -235,18 +254,17 @@ def flush(self):
         self.__http.putheader('Host', self.host)
         self.__http.putheader('Content-Type', 'application/x-thrift')
         self.__http.putheader('Content-Length', str(len(data)))
-
-        if (not self.__custom_headers or
-                'User-Agent' not in self.__custom_headers):
+        if (not self._http_header_factory.get_headers() or
```

Comment:
the `self._http_header_factory.get_headers()` may have side effect, like someone's custom header factory may add a count on it, or have a performance concern like reading config from database.

So I think we should call it just once in this function and store it in a variable.

Reply:
that was a great catch, thanks/

I just pushed an update with fixes